internal/controller/streams.go

```diff
@@ -707,6 +707,12 @@ func ConnectStreams(ctx context.Context, logger *zap.Logger, envMap EnvironmentC
 			sm := &streamManager{
 				streamClient:       streamClient,
 				bufferedGrpcSyncer: bufferedGrpcSyncer,
+				FlowCache: FlowCache{
+					cache:      make(map[FlowKey]*list.Element),
+					queue:      list.New(),
+					bufferSize: 100,
+				},
+				flowChannel: make(chan interface{}, int(100)),
```

Comment:
100 is too large a channel IMHO.
```suggestion
				FlowCache: NewFlowCache(
					20*time.Second, // TODO: Make the active timeout configurable.
					1000, // TODO: Make the maxFlows capacity configurable.
					make(chan Flow, 10),
				)
```